### PR TITLE
Swagger Update for Microsoft.StorageSync (2018-07-01)

### PR DESCRIPTION
**Pull request by ankushbindlish2**
- Added TierFilesOlderThanDays in ServerEndpoint
- Add TriggerRollover request
- Remove Tags from ServerEndpoint

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json [hunk 1]
```diff
@@ -2508,15 +2682,211 @@
...
+            },
+            "description": "Gets or sets a list of key value pairs that describe the resource. These tags can be used for viewing and grouping this resource (across resource groups). A maximum of 15 tags can be provided for a resource. Each tag must have a key with a length no greater than 128 characters and a value with a length no greater than 256 characters."
         },
+        "properties": {}
+    },
+    "required": [
+        "location"
+    ],
+    "description": "The parameters used when creating a storage sync service."
+},
```
Comment (sergey-shandar) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: nit: spaces

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json [hunk 2]
```diff
@@ -2544,29 +2914,18 @@
...
-          "x-ms-client-flatten": true,
-          "$ref": "#/definitions/RegisteredServerProperties",
-          "description": "RegisteredServer properties."
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ProxyResource"
         }
-      },
-      "x-ms-azure-resource": true
+	  ],
```
Comment (sergey-shandar) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: nit: tab?

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json [hunk 3]
```diff
@@ -2544,29 +2914,18 @@
...
-        "type": {
-          "readOnly": true,
-          "type": "string",
-          "description": "Resource type"
-        },
-        "properties": {
-          "x-ms-client-flatten": true,
-          "$ref": "#/definitions/RegisteredServerProperties",
-          "description": "RegisteredServer properties."
+      "allOf": [
```
Comment (sergey-shandar) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: Do we need `allOf` here?

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json [hunk 4]
```diff
@@ -3190,6 +3547,12 @@
           "maximum": 100,
           "description": "Level of free space to be maintained by Cloud Tiering if it is enabled."
         },
+        "tierFilesOlderThanDays": {
+          "type": "integer",
```
Comment (sergey-shandar) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: This type is used several times. I would recommend to declare this type in `definitions` and then use `$ref`.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json [hunk 5]
```diff
@@ -2267,6 +2276,153 @@
...
+        ],
+        "operationId": "RegisteredServers_triggerRollover",
+        "description": "Recall a serverendpoint.",
+        "x-ms-examples": {
+          "RegisteredServers_triggerRollover": {
+            "$ref": "./examples/RegisteredServers_TriggerRollover.json"
+          }
+        },
+        "responses": {
+          "200": {
```
Comment (ravbhatnagar) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: Is there a response body on the successful completion of the rollover?

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json [hunk 6]
```diff
@@ -2508,15 +2682,211 @@
...
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SyncGroupCreateParametersProperties",
+          "description": "The parameters used to create the sync group"
+        }
+      },
+      "description": "The parameters used when creating a sync group."
+    },
+    "SyncGroupCreateParametersProperties": {
```
Comment (ravbhatnagar) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: Why is the property section empty? Are there no properties for this resource?

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json [hunk 7]
```diff
@@ -2508,15 +2682,211 @@
...
+      "properties": {
+        "storageAccountResourceId": {
+            "type": "string",
+            "description": "Storage Account Resource Id"
+        },
+        "storageAccountShareName": {
+            "type": "string",
+            "description": "Storage Account Share name"
+        },
+        "storageAccountTenantId": {
```
Comment (ravbhatnagar) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: please remind me why is tenantId needed? can the storageAccount be in a different tenant?

Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: yup. You're correct. Cloudendpoint creation can point to different StorageAccount Tenant  Id

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json (hunk 6 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: Yes, No properties so far

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json (hunk 5 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: no response to this request

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json (hunk 4 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: @sergey-shandar Will WontFix work on this comment ? consistent with other properties defined ?

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json (hunk 3 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-07-01/storagesync.json]: yes , It's a proxy resource

**Review by sergey-shandar — APPROVED** [review]
We still need ARM approval.

**Comment by ravbhatnagar** [comment]
Looks good. signing off from ARM side,

**Pull request merged**